src/TwigComponent/doc/index.rst

```diff
@@ -840,6 +840,76 @@ The ``with`` data is what's mounted on the component object.
 .. note::
 
     Embedded components *cannot* currently be used with LiveComponents.
+    
+Twig Component syntax
+-----------------
+    
+.. versionadded:: 2.9
+
+    This syntax has been introduced in 2.9, and it's still experimental.
+    
+Twig Component came with a new syntax to ease the readability of your template:
+
+.. code-block:: html+twig
+
+    {# templates/some_page.html.twig #}
+    <twig:table :headers="['key', 'value']" :data="[[1, 2], [3, 4]]">
+        <twig:block name="th_class">
+            {{ parent() }}
+        </twig:block>
+         <twig:block name="td_class">
+            {{ parent() }}
+        </twig:block>
+        <twig:block name="footer">
+            <div class="data-table-footer">
+               My footer
+            </div>
+        </twig:block>
+    </twig:table>
+    
+You can now use a custom tag prefix starting by <t: and your component name
+
+.. code-block:: html+twig
+
+    <twig:table></:table>
+    // and also self closing tag
+    <twig:table/>
+    
+You can pass an attribute to your component like any HTML attribute and also use the ':' prefix to your attribute to indicate that the value should be compiled by Twig
+
+.. code-block:: html+twig
+
+    // primary will be set to true
+    <twig:component primary></:component>
+    <twig:component text="hello" :user="user.id"/>
+    
+    // equal to
+    <twig:component text="hello" user="{{ user.id }}"/>
+    
+    // and pass object, or table, or anything you imagine
+    <twig:component :table="['col' => ['foo', 'oof']]"/>
```

Comment:
I would *first* show a normal attribute example - one that does not use a variable - explain how that passes in a prop. THEN have a second code block where you say that you can also pass in dynamic variables with 2 different syntaxes.

Reply:
Also, use at least one camelCase prop name to show that this isn't special.